src/main/java/com/google/sps/servlets/GrantLocationServlet.java

```diff
@@ -18,8 +19,16 @@
   */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    
-    //TODO - implement the rest of this function (@Kaleab)
+ 
+    final String correlator = Shared.getCorrelator(request);
+
+    if (correlator.isEmpty()) {
+      response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+      return;
+    }
+
+    Constants.updateDatabase(correlator, Shared.GRANTS_LOCATION);
```

Comment:
I like this change. I wonder if your files are out of date.  Did you call this Shared?

Reply:
Oops good catch!